engines/director/lingo/xlibs/spacemgr.cpp

```diff
@@ -0,0 +1,670 @@
+					result += "\r";
+					result += Common::String::format("VIEW %s", me->_curView.c_str());
+				}
+			}
+		}
+	}
+
+	if (debugLevelSet(5)) {
+		Common::String format = result;
+		for (int i = 0; i < (int)format.size(); i++) {
+			if (format[i] == '\r')
+				format.replace(i, 1, "\n");
+		}
+		debugC(5, kDebugXObj, "SpaceMgr::m_getCurData(): %s", format.c_str());
+	}
+
+	g_lingo->push(Datum(result));
+}
+
+void SpaceMgr::m_setCurData(int nargs) {
+	g_lingo->printSTUBWithArglist("SpaceMgr::m_setCurData", nargs);
+	if (nargs != 4) {
+		warning("SpaceMgr::m_setCurData: expected 1 argument");
+		g_lingo->dropStack(nargs);
+		g_lingo->push(Datum(0));
+		return;
+	}
+	SpaceMgrXObject *me = static_cast<SpaceMgrXObject *>(g_lingo->_currentMe.u.obj);
+	g_lingo->printSTUBWithArglist("SpaceMgr::m_setCurData", nargs);
+	Datum view = g_lingo->pop();
+	Datum node = g_lingo->pop();
+	Datum space = g_lingo->pop();
+	Datum spaceCollection = g_lingo->pop();
+	if (spaceCollection.type != STRING ||
+		space.type != STRING ||
+		node.type != STRING ||
+		view.type != STRING) {
+		warning("SpaceMgr::m_setCurData: expected args to be a string");
+		g_lingo->push(Datum(0));
+		return;
+	}
+	me->_curSpaceCollection = *spaceCollection.u.s;
+	me->_curSpace = *space.u.s;
+	me->_curNode = *node.u.s;
+	me->_curView = *view.u.s;
+	g_lingo->push(Datum(0));
+}
+
+void SpaceMgr::m_addSpaceCollection(int nargs) {
+	g_lingo->printSTUBWithArglist("SpaceMgr::m_addSpaceCollection", nargs);
+	g_lingo->dropStack(nargs);
+	g_lingo->push(Datum(0));
+}
+
+void SpaceMgr::m_removeSpaceCollection(int nargs) {
+	g_lingo->printSTUBWithArglist("SpaceMgr::m_removeSpaceCollection", nargs);
+	g_lingo->dropStack(nargs);
+	g_lingo->push(Datum(0));
+}
+
+void SpaceMgr::m_setCurSpaceCollection(int nargs) {
+	if (nargs != 1) {
+		warning("SpaceMgr::m_setCurSpaceCollection: expected 1 argument");
+		g_lingo->dropStack(nargs);
+		g_lingo->push(Datum(0));
+		return;
+	}
+	SpaceMgrXObject *me = static_cast<SpaceMgrXObject *>(g_lingo->_currentMe.u.obj);
+	Datum spaceCollection = g_lingo->pop();
+	if (spaceCollection.type != STRING) {
+		warning("SpaceMgr::m_setCurSpaceCollection: expected spaceCollection to be a string, not %s", spaceCollection.type2str());
+		g_lingo->push(Datum(0));
+		return;
+
+	}
+	if (!spaceCollection.u.s->empty())
+		me->_curSpaceCollection = *spaceCollection.u.s;
+	debugC(5, kDebugXObj, "SpaceMgr::m_setCurSpaceCollection: %s", spaceCollection.u.s->c_str());
+	g_lingo->push(Datum(0));
+}
+
+void SpaceMgr::m_getCurSpaceCollection(int nargs) {
+	if (nargs != 0) {
+		warning("SpaceMgr::m_getCurSpaceCollection: expected 0 arguments");
+		g_lingo->dropStack(nargs);
+	}
+	SpaceMgrXObject *me = static_cast<SpaceMgrXObject *>(g_lingo->_currentMe.u.obj);
+	Common::String result;
+	if (!me->_curSpaceCollection.empty()) {
+		if (me->_spaceCollections.contains(me->_curSpaceCollection)) {
+			result = Common::String::format("SPACECOLLECTION %s", me->_curSpaceCollection.c_str());
```

Comment:
You could write 
```C
result = "SPACECOLLECTION  " + me->_curSpaceCollection;
```

The same goes for the other similar cases in this file.

Reply:
Fixed